### DISCUSSION TITLE
html_body is not working in API

### DISCUSSION THREAD

**Discussion by rccess** [postalserver/postal#2313] [Help with using Postal]
i am using API code in PHP

plain_body working fine but when same code running with html_body , API return error

Message-Id: {"status":"internal-server-error"}

when i checked logs, found this error

invalid byte sequence in UTF-8

<img width="540" alt="image" src="https://user-images.githubusercontent.com/32220165/210920059-a63093ab-d42b-46f8-8d68-1fb5285d3ebc.png">

can you please advise on the same

Thanks

**Comment by willpower232** [comment]
are you able to provide a copy of the html_body you sent?

**Comment by rccess** [comment]
https://pastebin.com/yDN7dkQP

**Comment by willpower232** [comment]
Looks like you have a few odd characters in the message

![image](https://user-images.githubusercontent.com/1619102/211032527-2eaa9c0d-ca9e-4a88-a386-05bc2882025d.png)

**Comment by rccess** [comment]
no that not a  problem because when we replace " with /" that is working

https://pastebin.com/XaCRE8rJ

another concern is with /api/v1/send/raw

{
    "status": "error",
    "time": 0.0,
    "flags": {},
    "data": {
        "code": "FromAddressMissing",
        "message": "There was no valid From address found in this email."
    }
}

<img width="465" alt="image" src="https://user-images.githubusercontent.com/32220165/211033959-afe3101e-804e-4e67-ad3d-915600a8f0e7.png">

since from domain is proper verified and working with smtp credential under same smtp server

Send as any also enabled on the same server

can you please advise on the same ?

**Reply by willpower232** [reply]
as per the documentation, http://apiv1.postalserver.io/controllers/send/raw, when using the raw endpoint you have to firstly base64 encode the message and secondly it has to be formatted as an email with all the headers
